### DISCUSSION TITLE
Adding functionality to CSK app

### DISCUSSION THREAD

**Discussion by TomerErezSick** [SICKAppSpaceCodingStarterKit/.github#4] [Q&A] [answered]
hey there, Tomer Erez here, a Sick developer.
i am trying to add functions to one of the CSKs and struggling so far.
in the checkAPIs.lua i am adding a crown (connector.power for example) to the loadAPIs function.
then, when trying to use the functions from that crown in other scripts in my app , the console says Runtime error: attempt to index a nil value.

i was wondering what is the correct way to add crowns and functions to the csk then.

**Comment by golluroSICKAG** [comment, marked as answer]
Hi Tomer
sorry for the late reply!
Did you check to uncomment following line within the main script?

-------------------------------------------
-- If app property "LuaLoadAllEngineAPI" is FALSE, use this to load and check for required APIs
-- This can improve performance of garbage collection

-- _G.availableAPIs = require('Mainfolder/Subfolder/helper/checkAPIs') -- can be used to adjust function scope of the module related on available APIs of the device
-------------------------------------------

Please also see: https://github.com/SICKAppSpaceCodingStarterKit/.github/blob/main/docu/SICKAppSpaceCodingStarterKit_Documentation.md#app-parameter-lualoadallengineapi

BR
